app/views/shared/_alerts.html.erb

```diff
@@ -4,6 +4,9 @@
       <div class="alert alert-<%= klass %>">
         <button type="button" class="close" data-dismiss="alert">&times;</button>
         <%= message %>
+        <% if display_access_request_link?(type) %>
+          <%= link_to_request_access %>
+        <% end %>
```

Comment:
Sorry @msufa , I was actually suggesting the solution highlighted in the Stackoverflow, unless someone else (i.e., @fneves, @henders , etc.) came up with a better solution or with an opinion against that solution. :-D

The reason why I was saying that I don't like to see HTML generated in the controllers is because, usually, it's not the safest option, as we may end up with XSS issues when generating HTML in the backend. But, in this case, you're probably 100% sure of what's going to be generated and what needs to be escaped (if anything). The other reason why I tend to dislike is because HTML should be rendered in the View. :)

I think the main advantages of having the link being returned as part of the flash message, from the UnauthorizedController, is that the alerts template will remain generic and all the logic regarding what to display in this specific message, and how to display it, will be centralised in one single place and we can keep using the default error types from Rails.